### PR TITLE
ServerSideRender: Allow using component without defined attributes

### LINKED ISSUE #7214: "attributes" prop should be optional in ServerSideRender
**Describe the bug**
If you try to invoke ServerSideRender without an `attributes` prop, you get an error.

**To Reproduce**
1. Create a block a block without attributes, rendered via a `ServerSideRender` component, like so: `<ServerSideRender block="my/block" />`
2. See error: "Error loading block: Invalid parameter(s): attributes"

**Expected behavior**
If `attributes` isn't provided, it should be ignored inside `ServerSideRender`.

**Workaround**
It is possible to work around this by passing an empty value into attributes, such as `<ServerSideRender block="my/block" attributes={{}} />` (and then obviously just ignore it in your `render_callback`).

### PR DESCRIPTION
**Pull request by dimofte**
## Description

Fixes #7214
Optional `attributes` in `ServerSideRender` 

## How has this been tested?
Manually in Chrome on MacOS. 
The component doesn't have automated tests and I haven't implemented any.

## Types of changes
Arguably a bug fix or a non-breaking api change

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation.   - not applicable

### CODE REVIEW

**Comment by chrisvanpatten** [comment]
Looks like the Travis failure is unrelated; some files missing that aren't related to this PR.

File: components/server-side-render/index.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ export class ServerSideRender extends Component {
 		if ( null !== this.state.response ) {
 			this.setState( { response: null } );
 		}
-		const { block, attributes } = props;
+		const { block, attributes = {} } = props;
```
Comment (danielbachhuber) [inline components/server-side-render/index.js]: Should `attributes` be present in the request if it's empty?

**Comment by chrisvanpatten** [comment]
@danielbachhuber That's a great point. We took this approach for a quick solution with a small surface area, but it could definitely be improved.

Maybe if #7232 can be merged in its current state, we could close this PR and address it as part of the [longer-term fix](https://github.com/WordPress/gutenberg/pull/7232#issuecomment-396933372) mentioned in #7232? If we could use a JS implementation of `http_build_query` like I mentioned in that issue, we could just have the whole query string generated in JS rather than just tacking on the attributes.

**Review by danielbachhuber — APPROVED** [review]

**Comment by danielbachhuber** [comment]
I'm fine to land this as-is, given the developer-facing API won't change with the future enhancement.

**Comment by danielbachhuber** [comment]
Ignoring the `codecov` failure because it seems to be useless / false-positive.

**Pull request merged**